x/lockup/keeper/lock.go

```diff
@@ -66,100 +66,76 @@ func (k Keeper) AddToExistingLock(ctx sdk.Context, owner sdk.AccAddress, coin sd
 // Tokens locked are sent and kept in the module account.
 // This method alters the lock state in store, thus we do a sanity check to ensure
 // lock owner matches the given owner.
-func (k Keeper) AddTokensToLockByID(ctx sdk.Context, lockID uint64, owner sdk.AccAddress, coin sdk.Coin) (*types.PeriodLock, error) {
+func (k Keeper) AddTokensToLockByID(ctx sdk.Context, lockID uint64, owner sdk.AccAddress, tokensToAdd sdk.Coin) (*types.PeriodLock, error) {
 	lock, err := k.GetLockByID(ctx, lockID)
+	if err != nil {
+		return nil, err
+	}
 
 	if lock.GetOwner() != owner.String() {
 		return nil, types.ErrNotLockOwner
 	}
 
+	lock.Coins = lock.Coins.Add(tokensToAdd)
+	err = k.lock(ctx, *lock, sdk.NewCoins(tokensToAdd))
 	if err != nil {
 		return nil, err
 	}
-	if err := k.bk.SendCoinsFromAccountToModule(ctx, lock.OwnerAddress(), types.ModuleName, sdk.NewCoins(coin)); err != nil {
-		return nil, err
-	}
 
-	err = k.addTokenToLock(ctx, lock, coin)
-	if err != nil {
-		return nil, err
```

Comment:
This looks wrong?  addTokenToLock did more, e.g. handling of the main accumulation store? Why was this deleted?

Reply:
Oh the claim is that all functionality is there between the synth lock line, and lock.

Why not keep `addTokenToLock` and just have that function as a subcomponent call `lock`? (But retain the synth lock functionality)

That would be maintaining the standard in this package of having the exported method be doing validation logic & hook / return type logic, and the private method doing the state change

Reply:
I still don't fully follow the logic here. There are two calls that `addTokenToLock` makes that are not related to the synth lock related accumulation store update, which are:
1. `setLock` to update state with the new lock instance:
https://github.com/osmosis-labs/osmosis/blob/8f018a6c638b64912c1b22c7df04f84abe667ba0/x/lockup/keeper/lock.go#L100
2. Update to `accumulationStore`:
https://github.com/osmosis-labs/osmosis/blob/8f018a6c638b64912c1b22c7df04f84abe667ba0/x/lockup/keeper/lock.go#L106

How/where are these being accounted for in the new implementation?

Reply:
Oops, I see the `lock` call above now. The change that adds an addition parameter to `lock` also makes sense now. Clever refactor :)

Reply:
@ValarDragon  wdym by ?

> (But retain the synth lock functionality)